### PR TITLE
Add Lifetime documentation, and a convenience method for creating them

### PR DESCRIPTION
**Pull request by sharplet**

### CODE REVIEW

File: Documentation/FrameworkOverview.md [hunk 1]
```diff
@@ -108,6 +108,41 @@ from a [signal](#signals). Within RAC, an observer is represented as an [`Observ
...
+
+When observing a signal or starting a signal producer, it is important to consider how long the observation should last. For example, when observing a signal in order to update a UI component, it makes sense to stop observing it once the component is no longer on screen. This idea is expressed in ReactiveSwift by the `Lifetime` type.
+
+```swift
+import Foundation
+
+final class SettingsController {
+  // Define a lifetime for instances of this class. When an instance is
+  // deinitialized, the lifetime ends.
+  private let (lifetime, token) = Lifetime.makeLifetime()
```
Comment (mdiep) [inline Documentation/FrameworkOverview.md]: I didn't realize you could do this. TIL!

File: README.md [hunk 2]
```diff
@@ -49,6 +49,21 @@ let interrupter = frames.start { frame in ... }
 interrupter.dispose()
 ```
 
+#### `Lifetime`: limits the scope of an observation
```
Comment (eimantas) [inline README.md]: New line's missing after the heading.

File: README.md [hunk 3]
```diff
@@ -49,6 +49,21 @@ let interrupter = frames.start { frame in ... }
...
+  private let (lifetime, token) = Lifetime.makeLifetime()
+
+  func play() {
+    let frames: SignalProducer<VideoFrame, ConnectionError> = ...
+    frames.take(during: lifetime).start { frame in ... }
+  }
+}
+```
+
 #### `Property`: an observable box that always holds a value.
```
Comment (eimantas) [inline README.md]: New line's missing after the heading.

File: README.md (hunk 2 above)
Comment (sharplet) [inline README.md]: Yeah, I was going for consistency with the rest of the file.

**Pull request merged**